src/rrules/blas.jl

```diff
@@ -66,6 +66,12 @@ function arrayify(x::A, dx::DA) where {A,DA}
     return error(msg)
 end
 
+_complex(x::BlasRealFloat) = x
+_complex(x::Tangent{@NamedTuple{re::P, im::P}}) where {P<:BlasRealFloat} = complex(x.fields.re, x.fields.im)
+_cextract(x::Dual) = primal(x), _complex(tangent(x))
```

Comment:
Could we perhaps call all of these methods `numberify`, in order that they're analogous to `arrayify`? My feeling is that `_complex` and `_cextract` aren't the best name because they return `BlasRealFloat`s when the `Dual` is associated to real values.